chimney/src/main/scala/io/scalaland/chimney/internal/utils/DerivationGuards.scala

```diff
@@ -9,9 +10,19 @@ trait DerivationGuards {
 
   import c.universe._
 
-  def isScalaPBEnum(t: Type) = t.baseClasses.exists(_.fullName.contains("scalapb.GeneratedEnum"))
+  def isScalaPBEnum(t: Type): Boolean = t.baseClasses.exists(_.fullName.contains(ScalaPBGeneratedEnumBaseClassName))
 
-  def isScalaPBOneof(t: Type) = t.baseClasses.exists(_.fullName.contains("scalapb.GeneratedOneof"))
+  def isScalaPBOneof(t: Type): Boolean = t.baseClasses.exists(_.fullName.contains(ScalaPBGeneratedOneofBaseClassName))
+
+  def isSdlIdAnnotation(annotation: Annotation): Boolean =
+    annotation.tree.tpe.typeSymbol.fullName == SdlIdAnnotationFullName
+
+  def shouldThrowExOnMissingSdlId(sdlIdAnnotationParamValues: Set[String]): Boolean =
+    sdlIdAnnotationParamValues.contains(SdlIdGenerationManual)
+
+  def shouldUsePlaceholderOnMissingSdlId(sdlIdAnnotationParamValues: Set[String]): Boolean =
+    sdlIdAnnotationParamValues.contains(SdlIdGenerationAuto) ||
+      sdlIdAnnotationParamValues.contains(SdlIdGenerationTypeName) //Set contains name of the IdGeneration class if default param value was used
```

Comment:
I don't understand this condition

Reply:
it's Scala's reflection quirk

If the default value is used (`@id id: String`, instead of `@id(IdGeneration.Auto) id: String`), then the [reflection call](https://github.com/wix-incubator/chimney/pull/24/files#diff-8f6e2afc10259dcd205a31fb26f526ca79c6014c2524a9fef6265c823f5db457R162) will return `IdGeneration`, instead of `Auto`.
We have `Auto` as default in SDL, hence this check

Reply:
I see. Maybe worthwhile adding the example to the comment? I didn't understand simply from it (but well done on adding the comment from the get go)

Reply:
sure, will do in the next PR

Reply:
https://github.com/wix-incubator/chimney/pull/26/files#diff-0262f8c9239b2598fa7147d92db5b6d316c24e9139a0f15146cb16f4888aa5f1R26